atlas/AtlasMl/tests/routers/test_competency.py

```
@@ -150,3 +150,76 @@ def test_save_competencies_missing_required_fields(test_env, mock_weaviate_clien
+    course_id = ""
+    
+    response = client.get(
+        f"/api/v1/competency/relations/suggest/{course_id}",
+        headers={"Authorization": "secret-token"},
+    )
+    
+    # Should handle empty course_id (likely 404 or validation error)
+    assert response.status_code in [404, 422]
+
+
+def test_suggest_competency_relations_special_characters(test_env, mock_weaviate_client):
```

Ruff (0.12.2)

Line too long (89 > 88)

(E501)